esphome/components/ld2420/number/gate_config_number.cpp

```diff
@@ -0,0 +1,75 @@
+#include "gate_config_number.h"
+#include "esphome/core/helpers.h"
+#include "esphome/core/log.h"
+
+static const char *const TAG = "LD2420.number";
+
+namespace esphome {
+namespace ld2420 {
+
+void LD2420TimeoutNumber::control(float timeout) {
+  this->publish_state(timeout);
+  this->parent_->new_config.timeout = timeout;
+}
+
+void LD2420MinDistanceNumber::control(float min_gate) {
+  if ((uint16_t) min_gate > this->parent_->new_config.max_gate) {
+    min_gate = this->parent_->get_min_gate_distance_value();
+    this->publish_state(min_gate);
+  } else {
+    this->parent_->new_config.min_gate = (uint16_t) min_gate;
+    this->publish_state(min_gate);
+  }
```

Comment:
Could probably simplify this slightly:
```suggestion
    min_gate = this->parent_->get_min_gate_distance_value();
  } else {
    this->parent_->new_config.min_gate = (uint16_t) min_gate;
  }
  this->publish_state(min_gate);
```